models/transformer.py

```diff
@@ -144,13 +144,16 @@ def __init__(self, feature_name_to_idx, embedding_kwargs, use_masked_mha, num_la
     def call(self, x, mask, training):
         x = self.feature_embedding(x)
         x = self.dropout(x, training=training)
+        
+        attn_scores = []
         for i in range(self.num_layers):
-            x = self.enc_layers[i](x, mask, training)
+            x, attn_score = self.enc_layers[i](x, mask, training)
+            attn_scores.append(attn_score)
     
-        return x
+        return x, attn_scores
 
 class Transformer(tf.keras.Model):
-    def __init__(self, feature_name_to_idx, encoder_kwargs, decoder_kwargs):
+    def __init__(self, feature_name_to_idx, encoder_kwargs, decoder_kwargs, output_attn=False):
```

Comment:
I would propose to make `output_attn` configurable, so add it to `decoder_kwargs` in the model cfg and read it from there